pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -445,10 +452,17 @@ func (s *Scheduler) updateCachedPodData(p *corev1.Pod) {
 		Requests:           resources.RequestsForPods(p),
 		Requirements:       requirements,
 		StrictRequirements: strictRequirements,
+		IsDRAEnabled:       pod.HasDRARequirements(p),
 	}
 }
 
 func (s *Scheduler) add(ctx context.Context, pod *corev1.Pod) error {
+	// Check if pod has DRA requirements - if so, return DRA error when SkipDRAScheduling is enabled
+	if s.cachedPodData[pod.UID].IsDRAEnabled && karpopts.FromContext(ctx).FeatureGates.SkipDRAScheduling {
+		log.FromContext(ctx).V(1).WithValues("Pod", klog.KObj(pod)).Info("skipping pod scheduling, Dynamic Resource Allocation (DRA) is not yet supported by Karpenter")
```

Comment:
Since we're returning the error we don't need to add the log here. This will be logged here: https://github.com/kubernetes-sigs/karpenter/blob/218cca8f9a96046b356addb2d91cc42bf205e275/pkg/controllers/provisioning/scheduling/scheduler.go#L228-L229